File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.py
class Solution(object):
    def countConsistentStrings(self, allowed, words):
        st = set()
        for c in allowed:
            st.add(c)

        count = 0
        flag = True

        for s in words:
            flag = True
            for c in s:
                if c not in st:
                    flag = False
                    break
            
            if flag:
                count += 1
        
        return count


        

File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        set<char> st(allowed.begin(), allowed.end());
        int count = 0;
        bool flag = true;

        for(string s : words) {
            flag = true;
            for(char c : s) {
                if(st.find(c) == st.end()) {
                    flag = false;
                    break;
                }
            }

            if(flag) count++;
        }

        return count;
    }
};